packages/web-workers/src/messenger/i-frame/create-messenger.ts

```diff
@@ -1,27 +1,14 @@
 import {MessageEndpoint} from '@remote-ui/rpc';
 
 const RUN = '__run';
-const IFRAME_SOURCE = `
-  <html>
-    <body>
-      <script>
-        window.addEventListener('message', function (event) {
-          var data = event.data;
-          if (data == null || typeof data.${RUN} !== 'string') return;
 
-          var workerScript = URL.createObjectURL(
-            new Blob(['importScripts(' + JSON.stringify(data.${RUN}) + ');']),
-          );
-
-          var worker = new Worker(workerScript);
-          worker.postMessage({__replace: event.ports[0]}, [event.ports[0]]);
-        });
-      </script>
-    </body>
-  </html>
-`;
-
-export function createIframeWorkerMessenger(url: URL): MessageEndpoint {
+export function createIframeWorkerMessenger(
+  url: URL,
+  prepareIframe: (
+    iframe: HTMLIFrameElement,
+    identifier: string,
```

Comment:
I get why you are passing this through, but since it needs to be manually known for anyone using the non-null-origin iframe method anyways, I think I'd just drop this argument and hardcode it in the null origin file/ export it from a sibling `constants` file instead.